autoscraper/auto_scraper.py

```diff
@@ -17,37 +19,34 @@ def __init__(self, stack_list=None, url=None):
 
     def save(self, file_path):
         with open(file_path, 'w') as f:
-            f.write(json.dumps(self.stack_list))
+            json.dump(self.stack_list, f)
 
     def load(self, file_path):
         with open(file_path, 'r') as f:
-            self.stack_list = json.loads(f.read())
+            self.stack_list = json.load(f)
 
-    @staticmethod
-    def _get_soup(url=None, html=None, request_args=None):
+    @classmethod
+    def _get_soup(cls, url=None, html=None, request_args={}):
```

Comment:
using {} as default parameter value is not suggested and also raises a warning in some IDEs. 
https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python